test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs

```diff
@@ -84,57 +95,75 @@ public void LongCounterSerializationSingleMetricPoint(long value)
 
         var dataPoint = metric.Sum.DataPoints[0];
 
-        Assert.Equal(value, dataPoint.AsInt);
+        if (longValue != null)
+        {
+            Assert.Equal(longValue, dataPoint.AsInt);
+        }
+        else
+        {
+            Assert.Equal(doubleValue, dataPoint.AsDouble);
+        }
 
         // Assert time
         var metricPointsEnumerator = exportedItems[0].GetMetricPoints().GetEnumerator();
         metricPointsEnumerator.MoveNext();
         var metricPoint = metricPointsEnumerator.Current;
 
-        Assert.Equal((ulong)TimestampHelpers.ToUnixTimeNanoseconds(metricPoint.StartTime), dataPoint.StartTimeUnixNano);
+        Assert.Equal((ulong)metricPoint.StartTime.ToUnixTimeNanoseconds(), dataPoint.StartTimeUnixNano);
 
-        Assert.Equal((ulong)TimestampHelpers.ToUnixTimeNanoseconds(metricPoint.EndTime), dataPoint.TimeUnixNano);
+        Assert.Equal((ulong)metricPoint.EndTime.ToUnixTimeNanoseconds(), dataPoint.TimeUnixNano);
 
         AssertOtlpAttributes([new("tag1", "value1"), new("tag2", "value2")], dataPoint.Attributes);
     }
 
-    [Fact]
-    public void LongCounterSerializationMultipleMetricPoints()
+    [Theory]
+    [InlineData("longcounter", new long[] { 10, 20, 30 }, null)]
+    [InlineData("longcounter", new long[] { -10, 2, -30 }, null)]
+    [InlineData("doublecounter", null, new double[] { 10.20, 2, 30.65 })]
+    [InlineData("doublecounter", null, new double[] { -10.20, 2, -30.65 })]
+    public void CounterSerializationMultipleMetricPoints(string instrumentName, long[] longValues, double[] doubleValues)
```

Comment:
Add tests for `UpDownCounter<long>` and `UpDownCounter<double>` as well.

Reply:
coming up in one of the next PRs.